recipes-kernel/kernel-tests/kernel-tests-files/kernel_dmesg_diff.py

```diff
@@ -51,13 +51,24 @@ def get_device_desc():
     cmdoutput = run_cmd(['fw_printenv', 'DeviceDesc'])
     return cmdoutput.split('=')[1]
 
-class KernelVersion:
+class SemanticVersion:
+    def __init__(self, versionString):
+        # Regex below copied from https://semver.org/.
+        version = re.search(r'^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$', versionString)
+        self.full = version.group()
+        self.major = int(version.group('major'))
+        self.minor = int(version.group('minor'))
+        self.patch = int(version.group('patch'))
+        self.prerelease = version.group('prerelease')
+        self.build_metadata = version.group('buildmetadata')
+
+class KernelVersion(SemanticVersion):
     def __init__(self):
-        self.uname = run_cmd(['uname', '-r'])
-        kernel_version = re.search(r'((\d+\.\d+)\.\d+-rt\d+)(-next)?', self.uname)
-        self.full = kernel_version.group(1)
-        self.major_minor = kernel_version.group(2)
-        self.type = 'next' if kernel_version.group(3) else 'current'
+        version =run_cmd(['uname', '-r'])
+        SemanticVersion.__init__(self, version)
+        self.type = 'next' if self.build_metadata else 'current'
```

Comment:
Assuming that the NO_HZ kernel puts something unique in this field, then I guess this logic wouldn't work on that kernel or any other kernel variants we might dream up in the future. It's not a requirement now, so I don't think we necessarily need to change anything. Just wanted to point it out.